etc/scripts/docenizer.py

```diff
@@ -31,18 +34,31 @@
     'MOVQ',  # defined in MOVD:MOVQ
     'MOVSD'  # defined in MOVS:MOVSB:MOVSW:MOVSD:MOVSQ
 ]
+# Where to extract the asmdoc archive.
+ASMDOC_DIR = "asm-docs"
 
 
 class Instruction(object):
     def __init__(self, name, names, tooltip, body):
         self.name = name
+        self.url = "http://www.felixcloutier.com/x86/{}.html".format(urllib.quote(name))
         self.names = names
         self.tooltip = tooltip.rstrip(': ,')
         self.body = body
 
     def __str__(self):
         return "{} = {}\n{}".format(self.names, self.tooltip, self.body)
 
+def download_asm_doc_archive():
+    print("Downloading archive...")
+    urllib.urlretrieve("http://www.felixcloutier.com/x86/x86.zip", "x86.zip")
```

Comment:
What happens if the site is unavailable for some reason when we try to download the file?

Reply:
I guess it throws an exception. If the archive is not already downloaded and that there is no Internet, this seems like the behaviour I would expect. I could surround the statement by a try-catch block and cascade the error through the main (or try-catch in the main directly).

Reply:
Probably just pretty-printing a message explaining the error is the best approach, yes!

Reply:
I'm less enthusiastic about you guys downloading the whole archive in an automated test.

This stuff barely ever changes: the last update to Volume 2 was in September 2016. If you need the zip, consider caching it. If you just want to check that a page exist, the index page (index.html) has all the links to every other page in a very regular format.

Reply:
The automated tests are run only on generation. The tests will be run extremely rarely (as far as I understood) since they don't belong to the Continuous Integration. @mattgodbolt said he thinks they have regenerated may be twice.

If one wants to generate, it has to have the archive, which is either downloading and extracting manually the archive, or have the script do it.

From this point of view, is it okay with you if I let the script download the archive?

Reply:
If you're confident enough that it's the only reasonable thing to do and that I won't get angry emails from my host, it's probably fine. I would like you to be as efficient as possible with my bandwidth without preventing anything from happening. :)

Reply:
I'll leave the final decision to the maintainers, but judging from the contents generated by the docenizer, the file has as few reasons to change as the Intel manual ;) And fixes of course. 

It receives a lot of attention lately due to this PR but it should settle down once everything is corrected.

Reply:
Also, did you just change the formatting of some of your pages? 

I think the script now breaks because you've added some `<strong>` tags. 
If it's not you, then I've broken something :D

I can post you the diff later if you wish, I don't have it at hand right now.

Reply:
Some thresholds changed a bit. BeautifulSoup probably lets you extract text ignoring HTML, right? This is probably the most reliable.

It's a bit messy, but it's not even a tiny fraction of the mess that extracting text from a PDF is :)

Reply:
I have no idea how BeautifulSoup works, but I wouldn't expect less from a Python module ;)

I was mostly checking with you, to ensure I did not do something wrong.

I can't imagine the hassle you've been through. That's why I wouldn't do that, but I'm glad some crazy people are able to do it :smile: Congrats on succeeding, by the way.